osquery/remote/enrollment/enrollment_tests.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+#include <thread>
+#include <iostream>
+#include <boost/network/protocol/http/server.hpp>
+#include <osquery/enrollment.h>
+#include <gtest/gtest.h>
+
+#include "osquery/remote/requests.h"
+#include "osquery/remote/transports/http.h"
+#include "osquery/remote/serializers/json.h"
+
+namespace http = boost::network::http;
+
+namespace osquery {
+
+class EnrollmentTests : public testing::Test {
+ public:
+  void SetUp() {}
+};
+
+struct FooBarHTTPHandler;
+typedef http::server<FooBarHTTPHandler> Server;
+
+struct FooBarHTTPHandler {
+  void operator()(Server::request const &request, Server::response &response) {
+    response = Server::response::stock_reply(
+        Server::response::ok, std::string("{\"enrollment_key\":\"potatoes\"}"));
+  }
+  void log(...) {}
+};
+
+TEST_F(EnrollmentTests, test_enroll) {
+  // Call enroll
+  PluginRequest request = {
+      {"enroll", "1"}, // 0 enroll if needed, 1 force re-enroll
+  };
+  PluginResponse resp;
+  Status stat = Registry::call("enrollment", "enrollment", request, resp);
+  EXPECT_TRUE(stat.ok());
+  std::cout << stat.getMessage() << "\n\n\n\n\n";
+  // Verify get key contains the string
+  EXPECT_EQ(resp.size(), 1);
+  // EXPECT_EQ(resp[0]["key"], "potatoes");
+}
+}
+
+int main(int argc, char *argv[]) {
+  testing::InitGoogleTest(&argc, argv);
+  google::InitGoogleLogging(argv[0]);
```

Comment:
you may have to init osquery here (see core/init.cpp), but you can set `FLAGS_enrollment_uri` to be localhost here after you initialize flags.